diff --git a/tools/external_crates/google_metadata/Android.bp b/tools/external_crates/google_metadata/Android.bp
index 374dd6678..253c116fd 100644
--- a/tools/external_crates/google_metadata/Android.bp
+++ b/tools/external_crates/google_metadata/Android.bp
@@ -17,6 +17,7 @@ rust_library_host {
     edition: "2021",
     rustlibs: [
         "libchrono",
+        "libgoogle_metadata_proto",
         "libprotobuf",
         "libthiserror",
     ],
